src/v/raft/consensus.cc

```diff
@@ -4265,3 +4265,11 @@ size_t consensus::bytes_to_deliver_to_learners() const {
 }
 
 } // namespace raft
+
+namespace seastar {
+
+void lw_shared_ptr_deleter<raft::consensus>::dispose(raft::consensus* s) {
+    delete s;
+}
```

Comment:
why does Seastar require custom deleter if we are to use `shared_from_this`?

Reply:
it is a trick not to require the full type definition